marimo/_plugins/ui/_core/registry.py

```diff
@@ -71,22 +73,34 @@ def _has_parent_id(
                 return self._has_parent_id(element, parent_id)
         return False
 
-    def _register_bindings(self, object_id: UIElementId) -> None:
-        kernel = get_context().kernel
+    def _find_bindings_in_namespace(
+        self, object_id: UIElementId, glbls: Mapping[str, Any]
+    ) -> set[str]:
         # Get all variable names that are either:
         #   1. bound to this UI element, or
         #   2. bound to a view (child) of this element
-        names = set(
-            [
-                name
-                for name in kernel.globals
-                if (
-                    isinstance(kernel.globals[name], UIElement)
-                    and self._has_parent_id(kernel.globals[name], object_id)
-                )
-            ]
+        #
+        # Also introspects _Namespace objects, including the name of the
+        # _Namespace if it contains `object_id`
+        bindings: set[str] = set()
+        for name, value in glbls.items():
+            if isinstance(value, UIElement) and self._has_parent_id(
+                value, object_id
+            ):
+                bindings.add(name)
+            elif (
```

Comment:
This is the update that allows `set-ui-element-value` requests on objects inside a `_Namespace` returned by `cell.run()` to trigger reactive execution based on the containing `_Namespace` object.

There's a unit test in `test_runtime.py` that demonstrates this